noir-projects/noir-contracts/contracts/static_child_contract/src/main.nr

```diff
@@ -0,0 +1,136 @@
+// A contract used along with `StaticParent` contract to test static calls.
+contract StaticChild {
+    use dep::aztec::prelude::{AztecAddress, FunctionSelector, PublicMutable, PrivateSet, PrivateContext, Deserialize};
+
+    use dep::aztec::{
+        context::{PublicContext, Context, gas::GasOpts},
+        protocol_types::{abis::{call_context::CallContext}},
+        note::{note_getter_options::NoteGetterOptions, note_header::NoteHeader}
+    };
+    use dep::value_note::value_note::ValueNote;
+
+    #[aztec(storage)]
+    struct Storage {
+        current_value: PublicMutable<Field>,
+        a_private_value: PrivateSet<ValueNote>,
+    }
+
+    // Returns base_value + chain_id + version + block_number + timestamp statically
+    #[aztec(public)]
+    #[aztec(view)]
+    fn pub_get_value(base_value: Field) -> Field {
+        let return_value = base_value
+            + context.chain_id()
+            + context.version()
+            + context.block_number()
+            + context.timestamp() as Field;
+
+        return_value
+    }
+
+    // Sets `current_value` to `new_value`
+    #[aztec(public)]
+    fn pub_set_value(new_value: Field) -> Field {
+        storage.current_value.write(new_value);
+        context.emit_unencrypted_log(new_value);
```

Comment:
Why emit these logs?

Reply:
Another side effect we want to detect and prevent. Ideally we would have tests for each side effect, but it gets annoying fast (direct, parent-child, nested, individual side effect tests and then failure cases for each of them)